Query and update animals table

storres20 opened the pull request:
# In this milestone, I did next activities:
* Update 'animals' tables in SQL.
* Insert and query data in SQL.
* Search data by Query

storres20 committed:
update new data

storres20 committed:
update new queries

storres20 committed:
update schema

storres20 commented:
According to the exercises, I added a column species of type string to your animals' table.

After that, I typed some queries, and their results

```
1. Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
Inside a transaction:

    Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
    Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
    Commit the transaction.
    Verify that change was made and persists after commit.

Take a screenshot of the results of your actions. 
```
![Captura de pantalla de 2023-01-19 13-02-39](https://user-images.githubusercontent.com/81504385/213579480-55bd06a9-d6ea-4799-a243-e26a387fede6.png)

```
2. Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)
Take a screenshot of the results of your actions.
```
![Captura de pantalla de 2023-01-19 13-04-25](https://user-images.githubusercontent.com/81504385/213579703-0d17bb20-d00f-4732-b0f8-e5fef96e162d.png)

```
3. Inside a transaction:

    Delete all animals born after Jan 1st, 2022.
    Create a savepoint for the transaction.
    Update all animals' weight to be their weight multiplied by -1.
    Rollback to the savepoint
    Update all animals' weights that are negative to be their weight multiplied by -1.
    Commit transaction

Take a screenshot of the results of your actions.
```
![Captura de pantalla de 2023-01-19 13-44-08](https://user-images.githubusercontent.com/81504385/213579825-f0b90ff5-1d6e-489d-b28a-405c30a38d7c.png)

```
Write queries to answer the following questions: 
```
`How many animals are there?`
![Captura de pantalla de 2023-01-19 17-53-05](https://user-images.githubusercontent.com/81504385/213580163-b7bb0dc0-06f4-4d98-9bcf-159f3b4002d0.png)

`How many animals have never tried to escape?`
![Captura de pantalla de 2023-01-19 17-53-47](https://user-images.githubusercontent.com/81504385/213580269-3bd448ce-3699-4920-8548-00b252f35a28.png)

`What is the average weight of animals?`
![Captura de pantalla de 2023-01-19 17-54-22](https://user-images.githubusercontent.com/81504385/213580341-50a2eabe-c2a0-4061-8d2b-98ef3702d02c.png)

`Who escapes the most, neutered or not neutered animals?`
![Captura de pantalla de 2023-01-19 17-54-59](https://user-images.githubusercontent.com/81504385/213580434-24cdcf2a-f346-4c61-b4df-ce759d21cb0d.png)

`What is the minimum and maximum weight of each type of animal?`
![Captura de pantalla de 2023-01-19 17-55-50](https://user-images.githubusercontent.com/81504385/213580592-32cbde5a-5893-4e3a-9e30-cf1929084a43.png)

`What is the average number of escape attempts per animal type of those born between 1990 and 2000?`
![Captura de pantalla de 2023-01-19 17-56-32](https://user-images.githubusercontent.com/81504385/213580681-99e895aa-48f4-4bc8-bbdc-593efb1db7c4.png)

andersonlebon commented on data.sql:
```diff
@@ -7,3 +7,19 @@ INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_
+
+INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (9, 'Boarmon', '2005-06-07', 7, true, 20.4);
+
+INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (10, 'Blossom', '1998-10-13', 3, true, 20.4);
```
- [ ] Kindly make sure you insert the correct data that was given in the requirements. Note that here you should insert the flowing data `Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.`(the weight should be 17.0 kg)

andersonlebon reviewed (changes requested):
## Hi @storres20 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![almost-there](https://user-images.githubusercontent.com/65068771/141811112-252360b9-757f-4279-bb3b-b7a2a4675ad3.gif)

## To highlight :champagne: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

storres20 committed:
Fix code reviewer's suggestion

storres20 commented:
I've fixed the weight_kg of "Blossom" pokemon from 20.4kg to 17.0kg

According to the exercises, I added a column species of type string to your animals' table.

After that, I typed some queries, and their results

```
1. Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
Inside a transaction:

    Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
    Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
    Commit the transaction.
    Verify that change was made and persists after commit.

Take a screenshot of the results of your actions. 
```
![Captura de pantalla de 2023-01-19 23-31-06](https://user-images.githubusercontent.com/81504385/213618236-d05f9071-a209-44db-9d3c-f6dcd2904554.png)

```
2. Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)
Take a screenshot of the results of your actions.
```
![Captura de pantalla de 2023-01-19 23-38-40](https://user-images.githubusercontent.com/81504385/213618510-402a9136-40ff-45c1-8a11-84aff25bf58c.png)

```
3. Inside a transaction:

    Delete all animals born after Jan 1st, 2022.
    Create a savepoint for the transaction.
    Update all animals' weight to be their weight multiplied by -1.
    Rollback to the savepoint
    Update all animals' weights that are negative to be their weight multiplied by -1.
    Commit transaction

Take a screenshot of the results of your actions.
```
![Captura de pantalla de 2023-01-19 23-49-45](https://user-images.githubusercontent.com/81504385/213619609-3b9aecbd-3ffd-451b-b749-acf9555663a1.png)

```
Write queries to answer the following questions: 
```

`How many animals are there?`
![Captura de pantalla de 2023-01-19 23-51-26](https://user-images.githubusercontent.com/81504385/213619787-8c7c155f-b295-4f48-8866-7865e5e6f787.png)

`How many animals have never tried to escape?`
![Captura de pantalla de 2023-01-19 23-52-36](https://user-images.githubusercontent.com/81504385/213619945-e54a63fe-1431-4b68-91b8-7d8ca3bbfb8b.png)

`What is the average weight of animals?`
![Captura de pantalla de 2023-01-19 23-54-07](https://user-images.githubusercontent.com/81504385/213620077-7c661e96-74d0-473c-bd40-f8b6644bbfc2.png)

`Who escapes the most, neutered or not neutered animals?`
![Captura de pantalla de 2023-01-19 23-55-04](https://user-images.githubusercontent.com/81504385/213620177-2beac9d8-44bb-4fae-a347-c06eb834e41a.png)

`What is the minimum and maximum weight of each type of animal?`
![Captura de pantalla de 2023-01-19 23-56-41](https://user-images.githubusercontent.com/81504385/213620334-adbb332d-9d8d-4905-a88f-73bfbc78fe85.png)

`What is the average number of escape attempts per animal type of those born between 1990 and 2000?`
![Captura de pantalla de 2023-01-19 23-57-54](https://user-images.githubusercontent.com/81504385/213620459-6571dc34-eea9-41cc-a8c4-01f3c7508b7a.png)

storres20 committed:
Fix savepoint

CollinsTatang reviewed (approved):
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 

### **Optional Changes**
```
Please remove the microverse logo from your README file.
- You can add yours if you have one.
```
![image](https://user-images.githubusercontent.com/25227022/165483662-7b06c812-4cc0-40b9-a84c-4729d4cf10c3.gif)



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
